choreolib/src/main/java/choreo/Choreo.java

```diff
@@ -431,4 +434,54 @@ public static <SampleType extends TrajectorySample<SampleType>> AutoFactory crea
         Optional.of(trajectoryLogger),
         requireNonNullParam(alliance, "alliance", "Choreo.createAutoFactory"));
   }
+
+  /**
+   * Creates an alert under the "Choreo" group.
+   *
+   * @param name The name of the alert
+   * @param type The type of alert
+   * @return an Alert published under the "Choreo" group
+   */
+  public static Alert alert(String name, Alert.AlertType type) {
+    return new Alert("Choreo", name, type);
+  }
+
+  /**
+   * Creates a {@link MultiAlert} under the "Choreo" group.
+   *
+   * @param textGenerator A function that accepts a list of infringements and returns an alert
+   *     message
+   * @param type The type of alert
+   * @return a MultiAlert published under the "Choreo" group
+   */
+  public static MultiAlert multiAlert(
+      Function<List<String>, String> textGenerator, Alert.AlertType type) {
+    return new MultiAlert(textGenerator, type);
+  }
+
+  /**
+   * An alert that allows multiple "infringements", or error scenarios. Effectively, it merges
+   * multiple similar alert blurbs into one.
+   */
+  public static class MultiAlert extends Alert {
+    private final Function<List<String>, String> textGenerator;
+    private final List<String> infringement = new ArrayList<>();
+
+    MultiAlert(Function<List<String>, String> textGenerator, AlertType type) {
+      super("Choreo", textGenerator.apply(List.of()), type);
+      this.textGenerator = textGenerator;
+    }
+
+    /**
+     * Adds an "infringement" to the list of infringements in this alert, and pushes the alert to
```

Comment:
The technical term in industry for this is "fault" rather than "infringement".

Reply:
I chose "infringement" instead of fault as multiple infringing values(choreo trajectories, invalid values, etc.) are causing a single fault(the alert) to appear. It could use a better name tho

Reply:
That'll work.